### PR TITLE
Team1 UCSBCurriculum Backend

### LINKED ISSUE #12: Implement UCSBCurriculumController - team 1 Issue 22
Look at the code from S21 proj-ucsb-courses-search.

There is a controller that allows the frontend to access results from the UCSBCurriculumService.

Bring that code into this code base, including testing.

- [x] Create UCSBCurriculumController file in the correct folder
- [x] Modeled after S21 code and works correctly
- [x] write tests for the controller
- [x] 100% test coverage
- [x] 100% mutation tests

### LINKED ISSUE #18: Implement UCSBCurriculumService - team 1
Look at the code from S21 proj-ucsb-courses-search.

There is a UCSBCurriculumService.

Bring that code into this code base, including testing.

Refer to the documentation at https://developer.ucsb.edu/.

If you need credentials; Prof. Conrad can supply these to your team (ask on Slack.)

You can also apply for an account directly: ask only for these APIs (which are all auto-approved). Asking for more than this could delay your request.

Academics: Academic Curriculums
Academics: Academic Quarter Calendar
Students: Student Record Code Lookups
Use the code from the team01 assignment as a model.

- [x] imported UCSBCurriculumService in the correct folder
- [x] credentials are working
- [x] Service is modeled and working correctly
- [x] write tests
- [x] 100% coverage tests
- [x] 100% on mutation tests

### PR DESCRIPTION
**Pull request by lukeyoffe**
# Overview

UCSBCurriculumService and UCSBCurriculumController + tests

# Issues Addressed

Closes #12 
Closes #18

### CODE REVIEW

**Review by asharma381 — APPROVED** [review]
Look good to me! Had a minor comment about `see-slack`

**Review by LucienLuc — APPROVED** [review]
LGTM!

**Review by nazryb — COMMENTED** [review]
lgtm

File: .env.SAMPLE [hunk 1]
```diff
@@ -3,4 +3,5 @@ GOOGLE_CLIENT_SECRET=see-instructions-in-readme
 ADMIN_EMAILS=phtcon@ucsb.edu
 MONGODB_URI=see-instructions-in-readme
 START_QTR=20221
-END_QTR=20222
\ No newline at end of file
+END_QTR=20222
+UCSB_API_KEY=see-slack
```
[the remark this answers is not in the document]
Comment (pconrad) [reply .env.SAMPLE:7]: I think see-instructions-in-readme is probably a better placeholder in the long run.

**Review by NickWang11235 — APPROVED** [review]

**Review by sidd328 — APPROVED** [review]
LGTM

**Review by asharma381 — APPROVED** [review]
Merged conflict issue fixed. Looks good to me!

**Review by andrewhlu — APPROVED** [review]
Looks good to me!

**Comment by andrewhlu** [comment]
This looks like a solid 10 points

**Pull request merged**